blues/airnote.live
# feat: add google analytics to airnote

# Problem Context

We want a better understanding of how many users are:

Visiting their dashboard (if at all)
* Frequency of visitation
* Relative frequency vs. Time of Purchase
* Additionally, a larger need of this is to be able to join this data with routing data (through devicename?) general airnote data vs individual airnote data

## Changes

Add a new `Analytics.svelte` component that will record every time a user lands on the site and changes pages and fire off an event to GA.

Pop that new component into the root `_layout.svelte` file so it's available throughout the whole app.

## Screenshot (if applicable)

![Screenshot 2023-10-19 at 2 01 41 PM](https://github.com/blues/airnote.live/assets/20400845/b9ae1e75-4050-4efe-a162-e7cc9cc24b83)

![Screenshot 2023-10-19 at 2 04 24 PM](https://github.com/blues/airnote.live/assets/20400845/ba2d6ba7-faa9-47b3-bb7d-69c474e017ae)

## Testing

Unit / integration / e2e tests?

Steps to test manually?

You can visit the PR link site and at least verify that the GA tag and custom code is there by inspecting the Elements tab in DevTools and searching for `google`.

Any other sorts of testing notes to include?

Since this is GA we're talking about, there's not a lot of ways to manually verify this works before going to production, instead we just have to release it, visit some Airnote dashboards and check back later to see what's registering.

Actually - here's a screenshot from GA with some early data courtesy of my Netlify deploy preview dashboard.

![Screenshot 2023-10-19 at 2 37 31 PM](https://github.com/blues/airnote.live/assets/20400845/5d562584-a92d-41be-86b4-4ba2b6585cbf)

## Any other related PRs

n/a

## Ticket(s)

https://trello.com/c/ugFwqSI3/15-metrics-analytics-google-analytics-for-individual-airnote-dashboards
